lib/utils/houchdb.js

```diff
@@ -0,0 +1,64 @@
+'use strict';
+
+const PouchW = require('pouchdb-wrapper');
+const _ = require('lodash');
+const defaultOptions = {
+    namespace: 'default',
+    port: 5984,
+    hostname: 'localhost',
+    protocol: 'http',
+    db: 'my-db',
+    plugins: []
+};
+
+const register = function(server, options, next) {
+    options = _.merge(defaultOptions, options);
+
+    // register PouchDB plugins
+    // this appears safe to do repeatedly, every time we call this function
+    _.forEach(options.plugins, (plugin) => {
+        PouchW.PouchDB.plugin(plugin);
+    });
+
+    let config;
+    if (options.path) {
+        config = {
+            path: options.path,
+            name: options.name
+        };
+    } else if (options.conn) {
+        config = {
+            name: options.name,
+            conn: options.conn
+        };
+    }
+
+    const db = new PouchW(config);
+    const pluginName = module.exports.attributes.name;
+    const exposableObj = {};
+    exposableObj[options.name] = db;
+    exposableObj.constructor = PouchW;
+
+    // throw error if namespace already exists
+    if (
+        server.plugins[pluginName] &&
+        server.plugins[pluginName][options.name]
+    ) {
+        const msg = 'PouchDB namespace `' + options.prefix + '` already exists';
+        next(new Error(msg));
+    }
+
+    // attempt to get info from the db
+    return db.info()
```

Comment:
Ok, I discussed this with the pouch folks on slack. @cdaringe: they claim that the following is a viable way to postpone logic until after the remote DB has successfully been created:
```
var newDB = new PouchDB(name, config);
newDB.info()
    .then(continue);
```
This means that the above code should work as expected, and should not cause any errors as you suggested. If you have some test cases to prove otherwise, let's dive in and figure it out!

Their exact words:
>nolanlawsonBOT [11:11 AM] 
>dwood: once the stateless constructor is in, the first method after the constructor will do the request

>nolanlawsonBOT [11:11 AM]
>e.g. info() or really any method

>daleharveyBOT [11:11 AM] 
>dwood, new PouchDB().info().then is a good way to ensure the db is successfully created